### PR TITLE
feat(preview): Add `Preview` Component

### PR DESCRIPTION
**Pull request by pepeladeira**
`Preview` component that applies a text snippet under the subject line in the inbox but is hidden from the main email content when it’s opened.

Closes https://github.com/zenorocha/react-email/issues/12

### CODE REVIEW

File: packages/preview/src/preview.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+type PreviewElement = React.ElementRef<'div'>;
+type RootProps = React.ComponentPropsWithoutRef<'div'>;
+
+export interface PreviewProps extends RootProps {
+  text: string;
```
Comment (bukinoshita) [inline packages/preview/src/preview.tsx:7]:
```suggestion
  children: string;
```

File: packages/preview/src/preview.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      {...props}
+    >
+      {text ? (
+        <>
+          {text}
+          {renderWhiteSpace(text)}
+        </>
+      ) : (
+        ''
+      )}
```
Comment (bukinoshita) [inline packages/preview/src/preview.tsx:31]: I don't think we need the condition here.

File: packages/preview/src/preview.tsx (hunk 1 above)
Comment (pepeladeira) [reply packages/preview/src/preview.tsx:7]: I'll do that!

File: packages/preview/src/preview.tsx (hunk 2 above)
Comment (pepeladeira) [reply packages/preview/src/preview.tsx:31]: good point

File: yarn.lock [hunk 3]
```diff
@@ -3445,7 +3450,7 @@ joycon@^3.0.1:
   resolved "https://registry.yarnpkg.com/joycon/-/joycon-3.1.1.tgz#bce8596d6ae808f8b68168f5fc69280996894f03"
   integrity sha512-34wB/Y7MW7bzjKRjUKTa46I2Z7eV62Rkhva+KkopW7Qvv/OSWBqvkSY7vusOPrNuZcUG3tApvdVgNB8POj3SPw==
 
-js-beautify@^1.6.12:
+js-beautify@1.14.6, js-beautify@^1.6.12:
```
Comment (bukinoshita) [inline yarn.lock:3453]: Can you rebase and then run `yarn`? This shouldn't be included.

File: yarn.lock [hunk 4]
```diff
@@ -1134,6 +1134,11 @@
     expect "^28.0.0"
     pretty-format "^28.0.0"
 
+"@types/js-beautify@1.13.3":
```
Comment (bukinoshita) [inline yarn.lock:1137]: Can you rebase and then run `yarn`? This shouldn't be included.

File: packages/preview/readme.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+![React Email img cover](https://react-email-assets.vercel.app/preview.png)
```
Comment (bukinoshita) [inline packages/preview/readme.md:1]:
```suggestion
![React Email preview cover](https://react-email-assets.vercel.app/preview.png)
```

File: yarn.lock (hunk 4 above)
Comment (pepeladeira) [reply yarn.lock:1137]: yes! sorry about that

Comment (pepeladeira) [reply yarn.lock:1137]: shouldn't https://github.com/zenorocha/react-email/pull/107/files be merged first?

File: yarn.lock (hunk 3 above)
Comment (pepeladeira) [reply yarn.lock:3453]: shouldn't https://github.com/zenorocha/react-email/pull/107 be merged first?

File: packages/preview/package.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@react-email/preview",
+  "version": "0.0.1",
+  "description": "Text that appears under the subject line in the inbox",
```
Comment (bukinoshita) [inline packages/preview/package.json:4]: Maybe this?

```suggestion
  "description": "A preview text that will be displayed in the inbox of the recipient",
```

File: packages/preview/readme.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+<div align="center"><strong>@react-email/preview</strong></div>
+<div align="center">Text that appears under the subject line in the inbox.</div>
```
Comment (bukinoshita) [inline packages/preview/readme.md:4]:
```suggestion
<div align="center">A preview text that will be displayed in the inbox of the recipient.</div>
```

File: yarn.lock (hunk 4 above)
Comment (bukinoshita) [reply yarn.lock:1137]: No worries 😊

File: packages/preview/readme.md (hunk 7 above)
Comment (pepeladeira) [reply packages/preview/readme.md:4]: I like it!

**Comment by bukinoshita** [comment]
I was testing this, and it works fine on Gmail, but Apple Mail wasn't working. I dig a bit to know more about it, and we actually need Preheader and Preview text — as they are different things (I thought they were the same). I also think we should avoid the `span` here(?) Since the `span` won't respect the width/height, only the font styles.

@zenorocha Do you know a better way to test this?

**Comment by pepeladeira** [comment]
@bukinoshita I delete the `yarn.lock file` and `node_modules folder`, run the `yarn` command and `@types/js-beautify@1.13.3` stills appears in `yarn.lock file`.

**Comment by pepeladeira** [comment]
@bukinoshita Could you try again? I added other types of white space and I received correctly from Apple Mail on my phone.

I also remove the span tag and change the logic

**Comment by zenorocha** [comment]
Confirmed that this is working well on Gmail and Superhuman.

CleanShot 2022-09-26 at 19 12 08@2x

CleanShot 2022-09-26 at 19 12 36@2x

**Review by zenorocha — APPROVED** [review]

**Pull request merged**